sklearn/utils/validation.py

```diff
@@ -780,8 +780,9 @@ def check_array(
     elif hasattr(array, "iloc") and hasattr(array, "dtype"):
         # array is a pandas series
         pandas_requires_conversion = _pandas_dtype_needs_early_conversion(array.dtype)
-        if pandas_requires_conversion:
-            # Set to None, to convert to a np.dtype that works with array.dtype
+        if isinstance(array.dtype, np.dtype):
+            dtype_orig = np.result_type(array.dtype)
+        else:
```

Comment:
I think having the comment is nice so we know what setting `dtype_orig=None` does.

```suggestion
        else:
            # Set to None to let array.astype work out the best dtype
```

Reply:
Indeed. I wasn't sure that the comment reflected what was actually happening. Added it back